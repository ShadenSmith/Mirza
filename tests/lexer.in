factorial: n -> 1 if < n 2 else * n (factorial (- n 1).
print (factorial 10).
make_adder: x -> : y -> + x y..
= add_10 (make_adder 10).
print (add_10 23).

