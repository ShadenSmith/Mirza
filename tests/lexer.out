<IDENTIFIER 'factorial'>
<COLON ':'>
<IDENTIFIER 'n'>
<ARROW '->'>
<INT '1'>
<IDENTIFIER 'if'>
<OPERATOR '<'>
<IDENTIFIER 'n'>
<INT '2'>
<IDENTIFIER 'else'>
<OPERATOR '*'>
<IDENTIFIER 'n'>
<LPAREN '('>
<IDENTIFIER 'factorial'>
<LPAREN '('>
<OPERATOR '-'>
<IDENTIFIER 'n'>
<INT '1'>
<RPAREN ')'>
<DOT '.'>
<IDENTIFIER 'print'>
<LPAREN '('>
<IDENTIFIER 'factorial'>
<INT '10'>
<RPAREN ')'>
<DOT '.'>
<IDENTIFIER 'make_adder'>
<COLON ':'>
<IDENTIFIER 'x'>
<ARROW '->'>
<COLON ':'>
<IDENTIFIER 'y'>
<ARROW '->'>
<OPERATOR '+'>
<IDENTIFIER 'x'>
<IDENTIFIER 'y'>
<DOT '.'>
<DOT '.'>
<OPERATOR '='>
<IDENTIFIER 'add_10'>
<LPAREN '('>
<IDENTIFIER 'make_adder'>
<INT '10'>
<RPAREN ')'>
<DOT '.'>
<IDENTIFIER 'print'>
<LPAREN '('>
<IDENTIFIER 'add_10'>
<INT '23'>
<RPAREN ')'>
<DOT '.'>
